### ISSUE TITLE
Add date format in front client

### ISSUE THREAD

**Issue by iero** [twentyhq/twenty#8053]
I would like to display a date field as the "year" only (ie 2016) and not a long format as "25 Oct, 2024 08:39"

An option setting the date format (ie %Y) could be great.

**Comment by ijreilly** [comment]
@Bonapara we could support this as a setting as we do for decimals for number fields (`fieldMetadata.settings` / [PR](https://github.com/twentyhq/twenty/pull/7340)): a date format can be picked in the field setting and will trigger a different display.
The date format only impacts the display but not the value stored in the database which should remain of datetime (so it is possible to go from a "year" only display (2024) to a full-date display (01/10/2024) without missing any data. The input component should not change in my opinion: even if the user has a "year" only display they still need to pick a full date. 
what do you think @Bonapara ?

(Another option I think would be a change on the table view directly a bit as it is doable in a gsheet, but we dont support this for now)

**Comment by Bonapara** [comment]
Thanks so much for the tech input @ijreilly! I think from a product perspective it would make more sense as a layout parameter than a data model setting.

As we should still require a full date input, the data format would not change in the database so that it would be more about layout but we don't have that capability yet.

**Comment by ijreilly** [comment]
interesting @Bonapara 
Agree it's more a layout matter than a setting but I thought the same for decimals. just to better understand what's the rationale behind considering decimals display as a setting matter and not a layout matter?

**Comment by Bonapara** [comment]
Are decimals just a cosmetic feature? Do we always store numbers in a database with X decimal even if we don't display them?

**Comment by iero** [comment]
I use decimal for example to display a version number ie "1.4" or a lat/long coordinate (ie -16.35) or an id (ie 2456). So for me it is very important to be able to setup a different number of decimals for each case

**Comment by iero** [comment]
> Thanks so much for the tech input @ijreilly! I think from a product perspective it would make more sense as a layout parameter than a data model setting.
> 
> As we should still require a full date input, the data format would not change in the database so that it would be more about layout but we don't have that capability yet indeed.

indeed, the date has to remain the same in database but has to display differently if for example I want to setup a "year of creation". It's quite often that I need to display only the year and not the rest. Usualy I setup the date as YYYY-01-01 in this case, and I just want to see YYYY

**Comment by Bonapara** [comment]
Okay, Felix and I agreed that it would make more sense to have them in layout settings. However, since we are far from implementing field level layout settings, we can establish a version 1 of layout settings in the data model. In the long term, if we introduce field-level layout settings, these data model layout settings would become the "default layout settings."

Here is a Figma of how this should behave:
![image](https://github.com/user-attachments/assets/3dabe9d3-efc9-4ea0-a3ce-42e5c5f8d763)

**Comment by iero** [comment]
Sounds awesome!

**Comment by ijreilly** [comment]
> Are decimals just a cosmetic feature? Do we always store numbers in a database with X decimal even if we don't display them?

@Bonapara yes exactly, which is why I feel we are in the exact same configuration here, the date format being a cosmetic feature. Hence the idea to store the desired date format in the fieldMetadata's settings. Are you good with that? 

Just a note regarding the design - I feel the "Custom" option expecting a date format is not accessible to non-tech users. 
When we implement this, we need to make sure to have a validation before saving it, as we can expect invalid values.

**Comment by Bonapara** [comment]
Regarding the design, @ijreilly, could we make the value check occur on input "focus out"?

**Comment by ijreilly** [comment]
@Bonapara should this also impact how dates are input? We got this requirement from discord:
![Image](https://github.com/user-attachments/assets/b942e035-ffd1-479d-a34f-46f1efba07ff)

I would say it is doable with our date pickers, but would like confirmation from @lucasbordeau as I think we ve had date picker issues before

**Comment by Bonapara** [comment]
@ijreilly I believe this is more related to the Experience `date&time` settings rather than the display of relative dates!

![Image](https://github.com/user-attachments/assets/188d0e99-9ac3-4eca-bf29-bded4d82ae06)

**Comment by guillim** [comment]
Technical input:
- [ ] frontend : adapt the dropdown options in the settings
- [ ] frontend : change the date display in inline mode
- [ ] frontend : change the date display in table mode
- [ ] backend : change the database settings `FieldMetadataDateSettings` from `displayAsRelativeDate?: boolean;` to an enum
- [ ] : write a migartion for the database settings

**Comment by ijreilly** [comment]
Just a note from the design: maybe we need to add a "precise" option in case the "default" display is "relative" but the user wants to show the precise date for a specific field ?

**Comment by Bonapara** [comment]
What do you think about:

- Full date
- Relative
- Year only
